src/virtual-machine.js

```diff
@@ -470,6 +470,17 @@ class VirtualMachine extends EventEmitter {
     }
 
     /**
+     * Repopulate the workspace with the blocks of the current editingTarget. This
+     * allows us to get around bugs like gui#413.
+     */
+    refreshWorkspace () {
+        if (this.editingTarget) {
+            this.emitWorkspaceUpdate();
+            this.runtime.setEditingTarget(this.editingTarget);
```

Comment:
runtime.setEditingTarget is responsible for checking which threads are running and putting the glows back on the right blocks